vscode/src/chat/agentic/CodyTool.ts

```diff
@@ -127,7 +127,7 @@ class FileTool extends CodyTool {
                 subTag: ps`name`,
             },
             prompt: {
-                instruction: ps`To retrieve full content of a codebase file`,
+                instruction: ps`To retrieve full content of a codebase file (DO NOT LOAD .env)`,
```

Comment:
```suggestion
                instruction: ps`To retrieve full content of a codebase file`,
```

Do we need to add this? This is too specific and only works for one use case, which might be better if we just filter it with the ContextFilter that this tool uses: https://sourcegraph.com/github.com/sourcegraph/cody/-/blob/vscode/src/commands/context/file-path.ts?L76

Reply:
I really agree fully to use ContextFilter as a deterministic approach. You can't rely 100% on a non-deterministic algorithm.

Reply:
> Do we need to add this?
Yes this seems to be necessary for AI to "remember" this particular instruction when it comes to applying it to this particular tool.

Reply:
And yes, fully agree that we need a deterministic approach **on top of** what I'm suggesting here.

Reply:
I don't think this is needed anymore after the prompt update for [NOTES] right? It works for me with only the NOTE prompt change, and it also worked when we ask it through memory without making any changes to this tool prompt.

Reply:
(Replied on Slack)

Reply:
https://sourcegraph.slack.com/archives/C05N5655M41/p1733446864815209?thread_ts=1733262102.734289&cid=C05N5655M41

Reply:
Hmm it worked for me when i tested it. 

What if you replace `[NOTES]` with `[RULES]` in the prompt instead. Does it work for you?

My issue with `NEVER LOAD .env` is that “LOAD” doesn’t seem to make sense here, and the hard code (.env) file name is not actually solving the issue. We also don't want the rule to only apply to the file tool.

Let me check with the team and see what other options we have here.

Reply:
> What if you replace [NOTES] with [RULES] in the prompt instead. Does it work for you?

I can try. The problem seems to be the fact those two pieces of prompts ("do not load sensitive info" and "here's how to use this tool") are too far, though. It seems hard for AI to associate those two far pieces of text together.

> My issue with NEVER LOAD .env is that “LOAD” doesn’t seem to make sense here, and the hard code (.env) file name is not actually solving the issue. We also don't want the rule to only apply to the file tool.

Would you suggest a different word? I don't see mentioning .env as "hard-coded" per se - it's simply mentioning an example (we can phrase it differently), and we use specific "hard-coded" examples elsewhere.

> Let me check with the team and see what other options we have here.

Sure. Once you have a PR with a better solution, I can take a look. For now, this is a hot fix to prevent people from casually encountering this common problem.